modules/utils/graph/Redif.cpp

```diff
@@ -0,0 +1,325 @@
+#include "Redif.hpp"
+
+
+namespace metric
+{
+	template <typename recType, class Metric, typename Tv>
+	Redif<recType, Metric, Tv>::Redif(std::vector<std::vector<Tv>> trainData, size_t nNeighbors /* = 10 */, size_t nIter /* = 15 */)
+	{
+		this->nNeighbors = nNeighbors;
+		xTrain = blaze::DynamicMatrix<Tv>(trainData.size(), trainData[0].size());
+		for (size_t i = 0; i < xTrain.rows(); i++)
+			for (size_t j = 0; j < xTrain.columns(); j++)
+				xTrain(i, j) = trainData[i][j];
+		blaze::DynamicMatrix<Tv> trainDe(trainData.size(), trainData[0].size());
+		trainModel(trainDe, nIter);
+	}
+
+
+	template <typename recType, class Metric, typename Tv>
+	void Redif<recType, Metric, Tv>::trainModel(blaze::DynamicMatrix<Tv> & xTrainDe, size_t nIter)
+	{
+		xTrainDe = xTrain;

```

Comment:
if we ignore content of `xTrainDe` maybe it is better to return it instead of `void`, WDYT?